packages/labextension/src/connection.ts

```diff
@@ -78,9 +78,13 @@ export class NeptuneConnection {
   };
 
   getNotebook = () => {
+
     return this
       .getAuthorizationHeader()
-      .then(authorizationHeader => {
+      .then((authorizationHeader) => {
+        if (!this.params.notebookId) {
```

Comment:
I think you can return `return Promise.resolve({} as INeptuneNotebook)` at the begining of `getNotebook` method. I don't think we need auth header to determine that we can't get anything useful from API.

Reply:
Fixed.